angularSTEP/src/app/recipe-page/recipe-page.component.ts

```diff
@@ -0,0 +1,49 @@
+import { Component } from '@angular/core';
+import { AngularFirestoreCollection, AngularFirestore } from '@angular/fire/firestore';
+import { ActivatedRoute , ParamMap} from '@angular/router';
+import { map } from  'rxjs/operators';
+import { Observable } from 'rxjs';
+
+interface Recipe {
```

Comment:
Okay so an interface is also what I had before. You will be able to see in my most recent commit that I have added Class and Converters. You don't need to make this change now and it can be made later. However, Firestore allows you to upload a class and also convert data into a class if you tell it how. It makes retrieval of data really easy. Then also you wouldn't need to declare the interface on every page. This is just info for now, but in the future we can migrate to this structure.